packages/KState/KState.js

```diff
@@ -0,0 +1,27 @@
+export default {
+  props: {
+    initialState: {
```

Comment:
This component is similar to [component-component](https://reacttraining.com/reach-ui/component-component/). However, it can be simplified to not use react way of thinking. I would:

1. remove `update` method in favor of using vue's reactivity directly
1. bind to lifecycle methods: e.g. `@mounted`
1. return the data directly:
1. rename this component to `Komponent`?
```javascript
export default {
  props: {
    data: {
      type: Object,
      required: true
    }
  },

  data () {
    return {
      ...this.data
    }
  },

  created () { this.$emit('created') },
  mounted () { this.$emit('mounted') },
  // more here...

  render () {
    return this.$scopedSlots.default({
      data: this.$data
    })
  }
}
```

```vue

<template>
  <KState :data="{ count: 0 }" v-slot="{ data }" @mounted="someMethod">
    <div>
      <KButton @click="data.count = data.count - 1">-</KButton>
      {{ data.count }}
      <KButton @click="data.count = data.count + 1">+</KButton>
    </div>
  </KState>
</template>
```

Reply:
Maybe Im not seeing the value here other than consolidating the state for child components. Are there some real world examples where we might use this?

Reply:
I've been having trouble testing the update function, so I'll try this approach out. Theres a similar example just after the state one in https://michaelnthiessen.com/renderless-components-5-wild-experiments "Declarative Lifecycle Components"